app/api/v1/mediator/manifest.py

```diff
@@ -1,53 +1,142 @@
-from typing import Any, Optional
-
-from fastapi import APIRouter, Body, Depends, status
+from typing import Any, Optional, Tuple
+import sys
+from fastapi import APIRouter, Body, Depends, HTTPException, status, Response
 
+from electionguard.group import ElementModQ, hex_to_q
 from electionguard.manifest import Manifest
 from electionguard.schema import validate_json_schema
+from electionguard.utils import get_optional
 
 from app.core.schema import get_description_schema
 
-from ..models import ValidateManifestRequest, BaseValidationResponse, ResponseStatus
+from ....core.repository import get_repository, DataCollection
+from ..models import (
+    ManifestQueryResponse,
+    ManifestSubmitResponse,
+    ValidateManifestRequest,
+    BaseValidationResponse,
+    ResponseStatus,
+)
 from ..tags import MANIFEST
 
 router = APIRouter()
 
+# TODO: multi-tenancy
+CLIENT_ID = "electionguard-default-client-id"
+
+
+@router.get("", tags=[MANIFEST])
+def get_manifest(manifest_hash: str) -> ManifestQueryResponse:
+    """Get an election manifest by hash"""
+    crypto_hash = hex_to_q(manifest_hash)
+    if not crypto_hash:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="manifest hash not valid"
+        )
+    try:
+        with get_repository(CLIENT_ID, DataCollection.MANIFEST) as repository:
+            query_result = repository.get({"manifest_hash": crypto_hash.to_hex()})
+            if not query_result:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Could not find manifest {manifest_hash}",
+                )
+
+            return ManifestQueryResponse(
+                status=ResponseStatus.SUCCESS,
+                manifest=query_result["manifest"],
```

Comment:
i'm not thrilled about sidestepping the typing but I didnt want to go through the boilerplate of generalizing the repository interface just yet so i think this is fine for now